x/concentrated-liquidity/tick.go

```diff
@@ -410,22 +416,37 @@ func (k Keeper) GetTickLiquidityNetInDirection(ctx sdk.Context, poolId uint64, t
 	for ; iterator.Valid(); iterator.Next() {
 		tickIndex, err := types.TickIndexFromBytes(iterator.Key())
 		if err != nil {
-			return []queryproto.TickLiquidityNet{}, err
+			return []queryproto.TickLiquidityNet{}, 0, sdk.Dec{}, err
 		}
 
 		tickStruct, err := ParseTickFromBz(iterator.Value())
 		if err != nil {
-			return []queryproto.TickLiquidityNet{}, err
+			return []queryproto.TickLiquidityNet{}, 0, sdk.Dec{}, err
 		}
 
 		liquidityDepth := queryproto.TickLiquidityNet{
 			LiquidityNet: tickStruct.LiquidityNet,
 			TickIndex:    tickIndex,
 		}
+
+		if zeroForOne {
+			if liquidityDepth.TickIndex < currentTick {
+				if liquidityDepth.LiquidityNet.IsNegative() {
+					startTickLiquidity = startTickLiquidity.Sub(liquidityDepth.LiquidityNet)
+				}
```

Comment:
We should be able to always subtract

Reply:
I think its actually dependent on direction we are traversing, where the sign of the tick liquidity we are looking for is dependent on the ZFO value

Reply:
That makes sense but why is it conditional on net value being non-negative?